client/src/pages/Profile.jsx

```diff
@@ -1,13 +1,207 @@
-import React from 'react'
+import {React, useEffect, useState} from 'react'
 import { useAuth0 } from "@auth0/auth0-react";
+import NoteCard from '../components/NoteCard';
+import SelectStatus from '../components/SelectStatus';
+import BookCard from '../components/BookCard';
+import CommentCard from '../components/CommentCard';
+import NotePopup from '../components/NotePopup'
 
 export default function Profile() {
   
-  const { user } = useAuth0();
+  const { isAuthenticated, user } = useAuth0();
+  //modal shows up
+  const [show, setShow] = useState(false)
+  //stores data from 3 tables
+  const [allActions, setAllActions] = useState([])
+  //dropdown menu
+  const [selectStatus, setSelectStatus] = useState("")
+
+  const [bookIds, setBookIds] = useState([]);
+  const [filteredBooks, setFilteredBooks] = useState([])
+
+  const [activeTab, setActiveTab] = useState('books');
+
+  //gets all infos from 3 tables
+  async function getUserAllActions(id) {
+
+    try {
+      const response = await fetch(`/api/profile/${id}`);
+
+      const allActions = await response.json();
+      setAllActions(allActions);
+
+    } catch (error) {
+      console.log(error.message)
+    }
+  }
+
+  useEffect(() => {
+    if(isAuthenticated) {
+      getUserAllActions(user.sub);
+    }
+  }, [isAuthenticated, user]); 
+
+  
+  useEffect(() => {
+    console.log("DATA", allActions);
+  }, [allActions]);
+
+  function handleSelect (event) {
+
+    const value = event.target.value;
+    setSelectStatus(value)
+
+    setActiveTab('books')
+  }
+
+
+  function getBookIdBySelectStatus(selectStatus, data){
+
+    let shelf_status; 
+
+    switch (selectStatus) {
+      case "read":
+        shelf_status = 0
+      break;
+      case "to-read":
+        shelf_status = 1
+      break;
+      case "currently-reading":
+        shelf_status = 2
+      break;
+    }
+    
+    const filteredData = data.filter((item) => item.shelf_status === shelf_status)
+    
+    const bookIds = filteredData.map(item => item.api_id)
+
+    return bookIds
```

Comment:
You can chain these functions since there's no need for the variables

```suggestion
    return filteredData = data
    	.filter((item) => item.shelf_status === shelf_status)
    	.map(item => item.api_id)
```